### PR TITLE
Move AuthService

### PR DESCRIPTION
**Pull request by rautenrieth-da**
This PR moves all ledger API authorization related code into a new bazel project for `package com.digitalasset.ledger.api.auth`.

No functionality has changed, this is purely moving code around.

### CODE REVIEW

**Comment by gerolf-da** [comment]
@rautenrieth-da: if we want on-X implementations to potentially customize the behavior, we would also need to publish the `ledger-api-auth` module to maven central.

File: ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Claims.scala [hunk 1]
```diff
@@ -88,8 +88,8 @@ case class Claims(claims: Seq[Claim]) {
 object Claims {
 
   /** A set of [[Claims]] that does not have any authorization */
-  def empty: Claims = Claims(List.empty[Claim])
+  def empty = Claims(List.empty[Claim])
```
Comment (stefanobaghino-da) [inline ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Claims.scala]: Is there any specific reason for this to be a `def`?

File: ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Claims.scala [hunk 2]
```diff
@@ -88,8 +88,8 @@ case class Claims(claims: Seq[Claim]) {
...
 
   /** A set of [[Claims]] that has all possible authorizations */
-  def wildcard: Claims = Claims(List[Claim](ClaimPublic, ClaimAdmin, ClaimActAsAnyParty))
+  def wildcard = Claims(List[Claim](ClaimPublic, ClaimAdmin, ClaimActAsAnyParty))
```
Comment (stefanobaghino-da) [inline ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Claims.scala]: Same here, looks immutable, I believe we can safely share this.

File: ledger/ledger-api-integration-tests/BUILD.bazel [hunk 3]
```diff
@@ -44,6 +44,7 @@ dependencies = [
     "//ledger-api/rs-grpc-akka",
     "//ledger-api/rs-grpc-bridge",
     "//ledger-api/testing-utils",
+    "//ledger/ledger-api-auth:ledger-api-auth",
```
Comment (stefanobaghino-da) [inline ledger/ledger-api-integration-tests/BUILD.bazel]:
```suggestion
    "//ledger/ledger-api-auth",
```
Not sure of why we don't do this also in the rest of the file, feel free to put it there as well. :slightly_smiling_face:

Comment (rautenrieth-da) [reply ledger/ledger-api-integration-tests/BUILD.bazel]: This would be a perfect task for our `./fmt.sh` script

**Pull request merged**